### PR TITLE
Repairs some of the ROCAUC tests

### PR DESCRIPTION
**Pull request by rebeccabilbro**
This PR addresses part of #252, with a focus on repairing the skipped and failing tests.

Changes include:
- removes the engineered `X`, `yb`, `ym` datasets and the `load_binary_data` and `load_multiclass_data` functions and switches to using the binary and multiclass datasets generated by `test_classifier/conftest.py`
- adds `@pytest.mark.usefixtures("binary", "multiclass")` to the `ROCAUCTests` class
- switches from testing with `MultinomialNB` to `GaussianNB`, since the `conftest.py` data includes negative values, which causes a sklearn error, since `MultinomialNB` is not suited to data with negative values.
- repairs `xfailed` tests: `test_multiclass_rocauc`, `test_rocauc_no_micro`, `test_rocauc_no_macro`, `test_rocauc_no_macro_no_micro`, `test_rocauc_no_classes`, and `test_decision_function_rocauc`.

Still not working:
- `test_binary_rocauc` still fails with the `IndexError: too many indices for array` error; I will have to continue investigating this one.

Questions:
- There are two tests currently marked as skipped and still to be implemented, [`test_rocauc_label_encoded`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_rocauc.py#L305) and [`test_rocauc_not_label_encoded`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_rocauc.py#L312); can anyone clarify what they had in might to be tested here?

### CODE REVIEW

**Comment by bbengfort** [comment]
Thank you for making these updates! I'm pretty stoked that ROCAUC is moving ahead, and these changes are all excellent enhancements. 

To answer your question: `test_rocauc_label_encoded` is supposed to test a target whose classes are digits (e.g. `[0,1,2]` instead of strings -- all the tests already do this, so I don't think another test is required here (so you can remove this test stub). The `test_rocauc_not_label_encoded` was supposed to test a target whose classes are strings (e.g. `['a', 'b', 'c']`) - this is kind of a throwback to the problems we were having with the `ConfusionMatrix` visualizer and label encoding, but I think it would still be a useful test to have.

**Comment by rebeccabilbro** [comment]
Got it @bbengfort, thanks & will do

File: tests/test_classifier/test_rocauc.py [hunk 1]
```diff
@@ -69,46 +51,20 @@ class FakeClassifier(BaseEstimator, ClassifierMixin):
...
-        # Return train/test splits
-        return tts(X, y, test_size=0.2, random_state=42)
-
-    def load_multiclass_data(self):
-        """
-        Returns the multiclass test data set.
-        """
-        raise NotImplementedError("Need to add multiclass data soon!")
-
     @pytest.mark.skip(reason="binary classifiers don't currently work as expected")
```
Comment (bbengfort) [inline tests/test_classifier/test_rocauc.py]: If you have the chance while updating the Windows tests, would you mind changing this from `skip` to `xfail`?

File: tests/test_classifier/test_rocauc.py [hunk 2]
```diff
@@ -130,29 +86,24 @@ def test_binary_rocauc(self):
         visualizer.poof()
         self.assert_images_similar(visualizer)
 
-    @pytest.mark.xfail(reason="see issue #315")
     def test_multiclass_rocauc(self):
```
Comment (bbengfort) [inline tests/test_classifier/test_rocauc.py]: This test requires the increased Windows tolerance for AppVeyor

File: tests/test_classifier/test_rocauc.py [hunk 3]
```diff
@@ -176,21 +127,17 @@ def test_rocauc_quickmethod(self):
         # TODO: impage comparison of the quick method
         roc_auc(model, data.data, data.target)
 
-    @pytest.mark.xfail(reason="see issue #315")
     def test_rocauc_no_micro(self):
```
Comment (bbengfort) [inline tests/test_classifier/test_rocauc.py]: This test requires the increased Windows tolerance for AppVeyor

File: tests/test_classifier/test_rocauc.py [hunk 4]
```diff
@@ -201,21 +148,17 @@ def test_rocauc_no_micro(self):
         visualizer.poof()
         self.assert_images_similar(visualizer)
 
-    @pytest.mark.xfail(reason="see issue #315")
     def test_rocauc_no_macro(self):
```
Comment (bbengfort) [inline tests/test_classifier/test_rocauc.py]: This test requires the increased Windows tolerance for AppVeyor

File: tests/test_classifier/test_rocauc.py [hunk 5]
```diff
@@ -258,21 +198,17 @@ def test_rocauc_no_macro_no_micro(self):
         visualizer.poof()
         self.assert_images_similar(visualizer)
 
-    @pytest.mark.xfail(reason="see issue #315")
     def test_rocauc_no_classes(self):
```
Comment (bbengfort) [inline tests/test_classifier/test_rocauc.py]: This test requires the increased Windows tolerance for AppVeyor

**Comment by bbengfort** [comment]
@rebeccabilbro thanks for making those changes! I will squash and merge when the tests pass!

**Pull request merged**